test/groups/metric.jl

```diff
@@ -102,12 +102,13 @@ invariant_metric_dispatch(::TestDefaultInvariantMetricManifold, ::RightAction) =
         Y = hat(SO3, Identity(SO3, e), [3.0, 4.0, 1.0])
 
         G = MetricManifold(SO3, lmetric)
-        @test inner(G, p, X, Y) ≈ dot(X, Diagonal([1.0, 2.0, 3.0]) * Y)
-        @test norm(G, p, X) ≈ sqrt(inner(G, p, X, X))
+        @test_broken inner(G, p, X, Y) ≈ dot(X, Diagonal([1.0, 2.0, 3.0]) * Y)
```

Comment:
We should check why these are broken and hopefully fix them

Reply:
These are... interesting failures. I've fixed two so far. The "exp τ-invariant" failures are caused by ODE method for `exp` defaulting to the induced basis. I don't know how to fix this. Perhaps there should be an automatic definition for that based on the `local_metric` that's already defined? Or maybe `local_metric` should just be defined for `InducedBasis` in the test file?

These "inner/norm" tests are broken because there is currently no `translate_diff` for the right action on `SpecialOrthogonal` with `TFVector` argument. I'm not sure whether this ever worked correctly?

Reply:
@sethaxen could you take a look at this issue?

Reply:
IIRC, the ODE method only ever worked for manifolds whose default representation was in a global chart. So it maybe could work for a manifold whose default representation is in embedded coordinates, but it should not be expected to. Most metric-related methods probably only work within charts as well, and `local_metric` likewise requires being in a chart or should take a basis. I haven't had the time to review this PR, but is that consistent with the failures you're seeing?

Reply:
I've changed all metric-related functions to work with respect to bases (not charts because they are not strictly needed to express coordinates of these metric functions). In this PR I try to be consistent about usage of coordinates and these are just two cases that are tricky to fix.